### DISCUSSION TITLE
Unable to support Bazel due to assumption in how module root is found

### DISCUSSION THREAD

**Discussion by clayne11** [99designs/gqlgen#2170] [Q&A]
### What happened?
When trying to build using Bazel, the environment is pretty different from regular `go build`. Specifically, the [goModuleRoot](https://github.com/99designs/gqlgen/blob/master/internal/code/imports.go#L58) function assumes that there will be a `go.mod` file which is not always true.

### What did you expect?
We need a way to set a package root or set of package roots manually.

Are you open to a PR for this?

**Comment by frederikhors** [comment]
Please PR.

**Reply by clayne11** [reply]
Sent a PR -- #2173

**Comment by clayne11** [comment]
I found a better solution. See #2173 for my comment describing what I did.
